### DISCUSSION TITLE
SERVER API : partial route doesn't seem to work

### DISCUSSION THREAD

**Discussion by thibautSchmutz** [nuxt/nuxt#18321] [Questions] [answered]
Hi every one, 

I have encountered an issue with my app. I created this function in *server / api / message.ts* :

```
import type { IncomingMessage, ServerResponse } from "http";

export default (req: IncomingMessage, res: ServerResponse) => {
  res.statusCode = 200;
  res.end("Message from Server");
};
```
Then, I tried to fetch it from *pages / index.vue* using the recommended *useAsyncData* as such : 

```
<template>
    <p>{{ message }}</p>  //  this does not show
</template>

<script setup>
  const { data: message } = await useAsyncData("res", () =>
    $fetch("/api/message")
  );
  onMounted(() => {
    console.log(message.value)  // this shows "Message from Server" in the console
  });
</script>
```
As you can see above, the console retrieves the data as it should but the template does not get it. So I tried to update to URL with the full path, as "http://localhost:3000/api/message" instead of just *"/api/message"*. And it fixes the problem ! 
I kept working on this path and I created an environment variable that would stock the chunk "http://localhost:3000". And I would use it every time I need to fetch from the server folder. 

Even though this walkaround seems to work, I am wondering if there is something I can do to make it work as it should ? 

Thank you and Vive Nuxt !

**Comment by HenrijsS** [comment, marked as answer]
Does the template just display an empty message?
Tried the same exact code on my end and worked without problems.

**Reply by thibautSchmutz** [reply]
It does ? It doesn't displays anything on mine. It may be a problem on my machine then... 
Thank you very much for having tried it :)  I know where to look for now

**Reply by HenrijsS** [reply]
Any console errors?
And make sure to update to the latest nuxt version: `yarn upgrade nuxt3`
They update it daily and it's changing all the time.

**Reply by thibautSchmutz** [reply]
No, I don't have any console errors either. the console give me the correct data ("Message from Server"). Yes, I will try to upgrade nuxt as it was a project I started the day the day they released the public beta

**Reply by thibautSchmutz** [reply]
I also tried to deploy it to Vercel without specifying the full URL and did not work either. So I changed to code back to using environement variable and it worked on the deployed version. I will get to the bottom of it and update this discussion if someone faces the same issue. 
Thanks for your help !

**Reply by thibautSchmutz** [reply]
I tried with another computer and it works just fine. It a local issue

**Reply by HenrijsS** [reply]
Another thing I noticed is that my browser caches some weird stuff.
Try to delete all site data for Localhost. Fixed a couple of issues for me.

**Reply by thibautSchmutz** [reply]
I've faced some issues too with localhost. it had conflicts between projects launch locally (console errors like 'component already assigned' or 'could not fetch route pages/...'.)

**Reply by HenrijsS** [reply]
To see if it's a browser issue, can you maybe create a quick GitHub repo?
I can check it on my end if it has any errors.
If not, I could create an issue here about browser caching incorrectly.

**Reply by thibautSchmutz** [reply]
Shoot, I deleted this repo... Since everything was fixed by creating a new project from scratch, I made space on my computer to not get confused later on. If I encounter this error again, I will push the repo and let you know.
